newton/_src/usd/schemas.py

```diff
@@ -0,0 +1,289 @@
+            "rotZ_position": SchemaAttribute("state:rotZ:physics:position", 0.0),
+            "angular_velocity": SchemaAttribute("state:angular:physics:velocity", 0.0),
+            "linear_velocity": SchemaAttribute("state:linear:physics:velocity", 0.0),
+            "rotX_velocity": SchemaAttribute("state:rotX:physics:velocity", 0.0),
+            "rotY_velocity": SchemaAttribute("state:rotY:physics:velocity", 0.0),
+            "rotZ_velocity": SchemaAttribute("state:rotZ:physics:velocity", 0.0),
+        },
+        PrimType.SHAPE: {
+            # Mesh hull vertex limit
+            "mesh_hull_vertex_limit": SchemaAttribute("physxConvexHullCollision:hullVertexLimit", 64),
+            # Collision contact offset
+            "rigid_contact_margin": SchemaAttribute("physxCollision:contactOffset", float("-inf")),
+        },
+        PrimType.MATERIAL: {
+            "stiffness": SchemaAttribute("physxMaterial:compliantContactStiffness", 0.0),
+            "damping": SchemaAttribute("physxMaterial:compliantContactDamping", 0.0),
+        },
+        PrimType.BODY: {
+            # Rigid body damping
+            "rigid_body_linear_damping": SchemaAttribute("physxRigidBody:linearDamping", 0.0),
+            "rigid_body_angular_damping": SchemaAttribute("physxRigidBody:angularDamping", 0.05),
+        },
+    }
+
+
+def solref_to_stiffness(solref):
+    """Convert MuJoCo solref (timeconst, dampratio) to internal stiffness.
+
+    Standard mode (timeconst > 0): k = 1 / (timeconst^2)
+    Direct mode (both negative): k = -timeconst (encodes -stiffness directly)
+    """
+    try:
+        timeconst = float(solref[0])
+        dampratio = float(solref[1])
+    except (TypeError, ValueError, IndexError):
+        return None
+
+    # Direct mode: both negative → solref encodes (-stiffness, -damping)
+    if timeconst < 0.0 and dampratio < 0.0:
+        return -timeconst
+
+    # Standard mode: k = 1 / (timeconst^2)
+    if timeconst <= 0.0:
+        return None
+
+    return 1.0 / (timeconst * timeconst)
+
+
+def solref_to_damping(solref):
+    """Convert MuJoCo solref (timeconst, dampratio) to internal damping.
+
+    Standard mode (both positive): b = 2 * dampratio / timeconst
+    Direct mode (both negative): b = -dampratio (encodes -damping directly)
+    """
+    try:
+        timeconst = float(solref[0])
+        dampratio = float(solref[1])
+    except (TypeError, ValueError, IndexError):
+        return None
+
+    # Direct mode: both negative → solref encodes (-stiffness, -damping)
+    if timeconst < 0.0 and dampratio < 0.0:
+        return -dampratio
+
+    # Standard mode: b = 2 * dampratio / timeconst
+    if timeconst <= 0.0 or dampratio <= 0.0:
+        return None
+
+    return (2.0 * dampratio) / timeconst
+
+
+class SchemaResolverMjc(SchemaResolver):
+    name: ClassVar[str] = "mjc"
+
+    mapping: ClassVar[dict[PrimType, dict[str, SchemaAttribute]]] = {
+        PrimType.SCENE: {
```

Comment:
There are a LOT more options on `MjcSceneAPI`, why do we only support a small subset?